### ISSUE TITLE
Returning a mock to be used with a response model raises a validation error

### ISSUE THREAD

**Issue by dbatten5** [tiangolo/fastapi#4364]
### First Check

- [X] I added a very descriptive title to this issue.
- [X] I used the GitHub search to find a similar issue and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/samuelcolvin/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
from typing import List
from unittest.mock import MagicMock, patch

from fastapi import FastAPI
from fastapi.testclient import TestClient
from pydantic import BaseModel


def get_objects():
    return [{"name": "bar"}]


class MyModel(BaseModel):
    name: str


app = FastAPI()


@app.get("/", response_model=List[MyModel])
def root():
    return get_objects()


client = TestClient(app)


@patch("{this_module_name}.get_objects")
def test_root(mock_get_objects):
    mock_object = MagicMock()
    mock_object.configure_mock(name="foo")
    mock_get_objects.return_value = [mock_object]

    response = client.get("/")

    assert response.json() == [{"name": "foo"}]
```

### Description

I'm trying to mock a call to an external library (in the above example this is simplified to calling the `get_objects` function) to return some mock objects which FastAPI then converts to a json response using the pydantic `response_model`. I would expect in the above example the `test_root` test to pass, however it fails with the following error:

```
pydantic.error_wrappers.ValidationError: 1 validation error for MyModel
response -> 0 -> name
  field required (type=value_error.missing)
```

If I run a test without mocking the call to `get_objects` like so:

```python
def test_real_root():
    response = client.get("/")
    assert response.json() == [{"name": "bar"}]
```

the test passes.

How can I pass some Mock objects as a return value for a route to be successfully converted to json using the `response_model`?

### Operating System

macOS

### Operating System Details

Catalina 10.15.7

### FastAPI Version

0.67.0

### Python Version

3.9.7

**Comment by dbatten5** [comment]
hi @harunyasar thanks for your reply. i think your example passes because you've changed the return value of `get_objects` to be a list of dicts, rather than a list of `unittest.MagicMock` objects. if i modify my example to return a list of dicts it does pass, just wondering if `response_model` is compatible working directly with `Mock` objects

**Comment by harunyasar** [comment]
> hi @harunyasar thanks for your reply. i think your example passes because you've changed the return value of `get_objects` to be a list of dicts, rather than a list of `unittest.MagicMock` objects. if i modify my example to return a list of dicts it does pass, just wondering if `response_model` is compatible working directly with `Mock` objects so i can leverage some of its useful features, eg. `spec`

I know what you mean. If you ask me this is not a valid question for `FastAPI` itself because I suspect it is highly relevant to `pydantic`. I am saying because `pydantic` only accepts dict values. At the end of the day, your data model is validated by a `pydantic` `BaseModel` class and it expects to get a dictionary to be validated. In short, a `Mock` object is not a valid model. In that case, you could simply pass a dict object.

**Comment by dbatten5** [comment]
yeah definitely possible it relates to `pydantic`, but then again fastapi does do some work to convert objects returned in route definitions into json using `response_model` (possibly by converting the object to a dict to pass to a `pydantic` `BaseModel`) so perhaps it could use some tweaking to work with `Mock/MagicMock` objects. i would prefer to work with `Mock` objects rather than `dicts` so i can use the `spec` feature and more closely mirror the actual objects i'm working with in my application
